### ISSUE TITLE
[1.12.2] Slow join/connect causing timeout with JEI installed

### ISSUE THREAD

**Issue by DeadSix27** [mezz/JustEnoughItems#1010]
So this is a frustrating issue, with `jei_1.12.2-4.7.11.102.jar` installed, joining a server takes very long and finally time's out.
But once I remove JEI, I can join the server just fine.. I've read in old issue reports something in the lines of: "mods can force JEI to reload" causing in mods reloading JEI again and again, increasing the join time but I think that got fixed... and I could not find that issue post again. (yet)
It just seemed very similar so I mentioned it.

~~I will test `jei_1.12.1-4.7.8.95.jar` now and get a game full log ready.~~ Same issue.

If you need it, here is the mod list: https://gist.github.com/DeadSix27/35ce97a26e9617e95c41e0071b6dd55f

EDIT: Possible relations: https://github.com/mezz/JustEnoughItems/issues/353 https://github.com/mezz/JustEnoughItems/issues/498

"Starting JEI took 1.952 min"
Log of client connecting successfully with JEI (rare): https://gist.github.com/DeadSix27/c3ff5b4f632ace951cd71e2d068bdf44

Log of server when a client joins and then times out: https://gist.github.com/DeadSix27/a39aaaefb3873b026fe3abd7ea44ad71

**Comment by mezz** [comment]
Thanks for the report!
Nothing should be able to cause JEI to reload like that any more, I'll take a look.

**Comment by mezz** [comment]
It looks like a few things are not working nicely together:
 * `twopiradians.blockArmor` has an incredible number of broken recipes
 * `rwtema.extrautils2` is logging lots of errors with `xreliquary`
Those caused about 20 seconds of slowdown.

`Sending runtime to plugin: twopiradians.blockArmor.jei.BlockArmorJEIPlugin took 18486 ms`
There's another 18 seconds from blockArmor.

In this log, `Starting JEI took 1.025 min` so it seems like those things took up the majority of the time.
Try removing blockArmor and extrautils2 to see what happens.

All of this happened before you even started the server though, it wouldn't prevent you from joining. JEI doesn't do much when players join the world anymore, and you would see "Restarting JEI." in the log if something restarted it. I'm not exactly sure why you are having an issue with that.

The only other weird thing I see is this:
```
[15:11:39] [main/INFO]: Logging in with username & password
[15:11:39] [main/ERROR]: Login failed:
com.mojang.authlib.exceptions.InvalidCredentialsException: Invalid credentials. Invalid username or password.
	at com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService.makeRequest(YggdrasilAuthenticationService.java:77) ~[YggdrasilAuthenticationService.class:?]
	at com.mojang.authlib.yggdrasil.YggdrasilUserAuthentication.logInWithPassword(YggdrasilUserAuthentication.java:78) ~[YggdrasilUserAuthentication.class:?]
	at com.mojang.authlib.yggdrasil.YggdrasilUserAuthentication.logIn(YggdrasilUserAuthentication.java:61) ~[YggdrasilUserAuthentication.class:?]
	at reauth.Secure.login(Secure.java:81) ~[Secure.class:?]
	at reauth.GuiLogin.login(GuiLogin.java:157) [GuiLogin.class:?]
	at reauth.GuiLogin.func_146284_a(GuiLogin.java:40) [GuiLogin.class:?]
	at net.minecraft.client.gui.GuiScreen.func_73864_a(GuiScreen.java:443) [blk.class:?]
	at reauth.GuiLogin.func_73864_a(GuiLogin.java:145) [GuiLogin.class:?]
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:533) [blk.class:?]
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:501) [blk.class:?]
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1757) [bib.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1096) [bib.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397) [bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_25]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_25]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_25]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_25]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
[15:11:46] [main/INFO]: Logging in with username & password
[15:11:46] [main/INFO]: Login successful!
```
Maybe it's caused by the ReAuth mod?

**Comment by DeadSix27** [comment]
@mezz I removed blockarmor and that seems to have removed a lot of joining time, in fact my friend (he lives across the globe) can now join without timing out. I noticed the issues blockarmor caused myself too but wanted confirmation. 

About extrautils2+xreliquary, is that something worth reporting on their tracker(s)? And  I do not think reauth is the issue, since removing that did not make a difference
